src/transformers/tokenization_utils.py

```diff
@@ -1534,71 +1602,151 @@ def _convert_encoding(
 
         return encoding_dict
 
-    def encode_plus(
-        self,
-        text,
-        text_pair=None,
-        return_tensors=None,
-        return_token_type_ids=True,
-        return_attention_mask=True,
-        return_overflowing_tokens=False,
-        return_special_tokens_mask=False,
-        **kwargs
-    ):
-        encoding = self.tokenizer.encode(text, text_pair)
-        return self._convert_encoding(
-            encoding,
-            return_tensors=return_tensors,
-            return_token_type_ids=return_token_type_ids,
-            return_attention_mask=return_attention_mask,
-            return_overflowing_tokens=return_overflowing_tokens,
-            return_special_tokens_mask=return_special_tokens_mask,
-        )
-
-    def tokenize(self, text):
-        return self.tokenizer.encode(text).tokens
-
     def _convert_token_to_id_with_added_voc(self, token):
-        id = self.tokenizer.token_to_id(token)
+        id = self._tokenizer.token_to_id(token)
         if id is None:
             return self.unk_token_id
         return id
 
     def _convert_id_to_token(self, index):
-        return self.tokenizer.id_to_token(int(index))
+        return self._tokenizer.id_to_token(int(index))
 
     def convert_tokens_to_string(self, tokens):
-        return self.decoder.decode(tokens)
+        return self._tokenizer.decode(tokens)
 
     def add_tokens(self, new_tokens):
-        self.tokenizer.add_tokens(new_tokens)
+        if isinstance(new_tokens, str):
+            new_tokens = [new_tokens]
+        return self._tokenizer.add_tokens(new_tokens)
 
     def add_special_tokens(self, special_tokens_dict):
         added = super().add_special_tokens(special_tokens_dict)
         self._update_special_tokens()
         return added
 
-    def encode_batch(
+    def num_added_tokens(self, pair=False):
+        # TODO: Need to expose PostProcessor.added_tokens, but it needs some refactoring on Rust side.
+        # Let's encode a special token so that it cant be truncated and just get the length - 1 or - 2 (pairs)
+        if pair:
+            return max(0, len(self.encode(self.mask_token or "", self.mask_token or "")) - 2)
+        else:
+            return max(0, len(self.encode(self.mask_token or "")) - 1)
+
+    def tokenize(self, text, **kwargs):
```

Comment:
This seems like it would take too much memory before being useful, as you really never have 2 sequences with 512/1024 tokens which are exactly the same